src/app/buy-credits/buy-credits.component.html

```diff
@@ -0,0 +1,150 @@
+                    <p><strong>Tip the Big Give</strong></p>
+                    <p class="tip-text">Big Give has a 0% platform fee. That means we're able to continue offering our services thanks to donors who leave an optional tip amount here:</p>
+                    <mat-form-field >
+                        <mat-label>Tip amount (%)</mat-label>
+                        <mat-select formControlName="tipPercentage" required="true" (selectionChange)="onTipSelectorChanged($event)">
+                            <mat-option [value]="7.5">7.5%</mat-option>
+                            <mat-option [value]="10">10%</mat-option>
+                            <mat-option [value]="12.5">12.5%</mat-option>
+                            <mat-option [value]="15">15%</mat-option>
+                            <mat-option [value]="'Other'">Other</mat-option>
+                        </mat-select>
+                    </mat-form-field>
+
+                    <div *ngIf="customTip()">
+                        <mat-form-field  class="form-field b-center size-lg">
+                            <mat-label>Custom Tip amount (£)</mat-label>
+                            <input matInput placeholder="Enter amount in £" formControlName="customTipAmount">
+                        </mat-form-field>
+                        <p class="error" *ngIf="customTipAmountField?.errors && customTipAmountField?.invalid && (customTipAmountField?.dirty || customTipAmountField?.touched)">
+                            Custom tips can range from £0 and {{ maximumDonationAmount | exactCurrency:'£' }} inclusive.
+                        </p>
+                    </div>
+
+                    <!-- Only show the calculated tip copy if the credit amount entered is valid AND if either a % tip is selected 
+                    from the dropdown OR a valid custom tip amount has been entered. In other words, this copy will not show if the credit
+                    amount entered is out of range, or invalid, or if the user clicks on 'Other' for the tip amount but has not yet
+                    specified how much the custom tip will be (i.e. blank custom tip), or has specified an invalid custom tip amount. -->
+                    <p *ngIf="
+                        creditAmountField?.value &&
+                        !creditAmountField?.errors &&
+                        !creditAmountField?.invalid &&
+                        !(customTip() && !customTipAmountField?.value) &&
+                        !customTipAmountField?.errors &&
+                        !customTipAmountField?.invalid
+                    "><strong>{{ calculatedTipAmount() | exactCurrency:'£' }}</strong> donation (tip) to support Big Give
+                        <span *ngIf="calculatedTipAmount() > 0">💗</span> <!-- display heart emoji if tip > £0 -->
+                        <span *ngIf="calculatedTipAmount() === 0 && customTip() && customTipAmountField?.value">😢</span> <!-- display sad emoji if custom tip amount set to £0 -->
```

Comment:
I'm not sure it's a good idea to be hitting big corporate donors with a sad face emoji!